File: android/app/src/main/java/com/kaspriv/wallet/room/WalletDao.java
package com.kaspriv.wallet.room;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.RawQuery;
import androidx.sqlite.db.SupportSQLiteQuery;
import java.util.List;

@Dao
public interface WalletDao {
    
    @Query("SELECT * FROM wallets")
    List<WalletEntity> getAllWallets();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertWallet(WalletEntity wallet);

    @Query("DELETE FROM wallets WHERE id = :id")
    void deleteWallet(String id);

    @Query("SELECT * FROM settings")
    List<SettingEntity> getAllSettings();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertSetting(SettingEntity setting);

    @Query("SELECT * FROM utxos")
    List<UtxoEntity> getAllUtxos();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertUtxo(UtxoEntity utxo);

    @Query("SELECT * FROM transactions")
    List<TransactionEntity> getAllTransactions();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertTransaction(TransactionEntity transaction);

    @RawQuery
    int executeRawWrite(SupportSQLiteQuery query);
}

File: android/app/src/main/java/com/kaspriv/wallet/room/WalletDao.kt
package com.kaspriv.wallet.room

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface WalletDao {
    
    @Query("SELECT * FROM wallets")
    fun getAllWallets(): List<WalletEntity>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertWallet(wallet: WalletEntity)

    @Query("DELETE FROM wallets WHERE id = :id")
    fun deleteWallet(id: String)

    @Query("SELECT * FROM settings")
    fun getAllSettings(): List<SettingEntity>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertSetting(setting: SettingEntity)

    @Query("DELETE FROM settings WHERE `key` = :key")
    fun deleteSetting(key: String)

    @Query("SELECT * FROM utxos")
    fun getAllUtxos(): List<UtxoEntity>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertUtxo(utxo: UtxoEntity)

    @Query("SELECT * FROM transactions")
    fun getAllTransactions(): List<TransactionEntity>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertTransaction(transaction: TransactionEntity)
}
